src/utils/Renderers.tsx

```diff
@@ -107,16 +107,84 @@ export const displayDebuggingFeatures = (
 //////////////////
 
 export const getMapMarkerColor = (
-  givenCombinedRow: Types.combinedRow,
-  currentCombinedRow: Types.combinedRow
+  givenRow: Types.CombinedRow,
+  currentRows: Types.CombinedRow[]
 ): string => {
-  const selectedSameCombinedRow = givenCombinedRow === currentCombinedRow;
-  return selectedSameCombinedRow
+  const selectedSameRow = givenInCurrentRows(givenRow, currentRows);
```

Comment:
Related to "Persisted row/marker highlight/details for combinedRows, including ones that split/merge upon zooming in/out"